### ISSUE TITLE
"Probabilities do not sum to 1" with NumPy backend

### ISSUE THREAD

**Issue by mlazzarin** [qiboteam/qibo#459]
**Describe the bug**
The script ``examples/variational_classifier/main.py`` raises a ValueError "probabilities do not sum to 1" when exectued with the NumPy backend. Last two lines of the traceback:
```
  File "../qibo/backends/numpy.py", line 232, in sample_frequencies
    frequencies = update_frequencies(nshots % SHOT_BATCH_SIZE, frequencies)
  File "../qibo/backends/numpy.py", line 224, in update_frequencies
    samples = self.random.choice(range(len(probs)), size=nsamples, p=probs)
  File "mtrand.pyx", line 939, in numpy.random.mtrand.RandomState.choice
ValueError: probabilities do not sum to 1
```
Maybe it is linked to this issue https://github.com/numpy/numpy/issues/6123.
Indeed, adding a normalization step for ``probs`` in the following function seems to solve the issue.
https://github.com/qiboteam/qibo/blob/9af4e773231374a83b3d0e93c47645d47f95cf27/src/qibo/backends/numpy.py#L218-L230

**To Reproduce**
1. Install ``qibo0.1.6rc1``
2. Execute the variational classifier example with the NumPy backend.

If needed, I managed to reproduce it in Google Colab.

**Comment by scarrazza** [comment]
@mlazzarin thank you very much for spotting this. Could you please retry with the branch #460?

**Comment by mlazzarin** [comment]
I confirm that the problem does not occur anymore with branch #460.

**Comment by scarrazza** [comment]
@mlazzarin thanks for checking.

**Comment by scarrazza** [comment]
Fixed in #460.
